### PR TITLE
[ehabi32] PACBTI-M unwinding rule in case of nested functions

### PR DESCRIPTION
**Pull request by sripar01**
In case of nested function, changes in the VSP value to be used as modifier for the Return Address Authentication.

@vhscampos

### CODE REVIEW

File: ehabi32/ehabi32.rst [hunk 1]
```diff
@@ -2652,9 +2652,21 @@ The encoding has been designed to meet the following criteria.
      address upon function exit. The Return Address Authentication Code is an
      unsigned integer with the same size as a general-purpose register. Upon
      completion of the unwinding instructions, the Return Address Authentication
-     Code can be used to validate the content of VRS[r15], using vsp as a
+     Code can be used to validate the content of VRS[r15] and if opcode 0xb5 is
+     encountered before completion of the unwinding, the effective VSP value at
+     code 0xb5 will be used as modifier otherwise the CFA is used as the
```
Comment (vhscampos) [inline ehabi32/ehabi32.rst:2657]: I'd add a comma before "otherwise"

File: ehabi32/ehabi32.rst [hunk 2]
```diff
@@ -2652,9 +2652,21 @@ The encoding has been designed to meet the following criteria.
...
      modifier.
 
+  h) In a very small number of cases (for example, nested functions when a
+     closure pointer is passed as an additional argument), the inter-procedural
+     scratch register (IP) may be live on entry to the function.  In this
```
Comment (vhscampos) [inline ehabi32/ehabi32.rst:2662]: Double space right before "In this"

File: ehabi32/ehabi32.rst [hunk 3]
```diff
@@ -2652,9 +2652,21 @@ The encoding has been designed to meet the following criteria.
...
+     situation the return address cannot be signed until IP has been copied
+     somewhere else (for example, by saving it onto the stack or copy to
```
Comment (vhscampos) [inline ehabi32/ehabi32.rst:2664]: "copy" -> "copying" to make it consistent

**Review by vhscampos — CHANGES_REQUESTED** [review]
A few sugestions

File: ehabi32/ehabi32.rst (hunk 3 above)
Comment (sripar01) [reply ehabi32/ehabi32.rst]: Statement related to argument register is removed due to a downstream comment.

**Review by vhscampos — APPROVED** [review]

**Pull request merged**